### PR TITLE
Corrected deposit all check amount logic

### PR DESCRIPTION
**Pull request by web3Liuwentong**
## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-cli/src/commands/deposit-all.ts [hunk 1]
```diff
@@ -227,13 +227,13 @@ export default class DepositAll extends IronfishCommand {
       const sendableOre = confirmedBalance - fee
 
       // terminate condition
-      if (terminate && pendingBalance < sendableOre + fee) {
```
Comment (web3Liuwentong) [inline ironfish-cli/src/commands/deposit-all.ts:230]: sendableOre + fee = confirmedBalance due to the upper logic, if pendingBalance < confirmedBalance didn't make sense.
so i change to if pendingBalance < fee then destory

File: ironfish-cli/src/commands/deposit-all.ts [hunk 2]
```diff
@@ -227,13 +227,13 @@ export default class DepositAll extends IronfishCommand {
...
+      if (terminate || pendingBalance < fee) {
         screen.destroy()
         process.exit(0)
       }
 
       // send transaction
-      if (confirmedBalance >= sendableOre + fee) {
```
Comment (web3Liuwentong) [inline ironfish-cli/src/commands/deposit-all.ts:236]: confirmedBalance = sendableOre + fee is always true due to line 227.

File: ironfish-cli/src/commands/deposit-all.ts [hunk 3]
```diff
@@ -227,13 +227,13 @@ export default class DepositAll extends IronfishCommand {
...
+      if (sendableOre >= minDepositOre) {
```
Comment (hughy) [inline ironfish-cli/src/commands/deposit-all.ts:236]: This change has already been applied in #2502

File: ironfish-cli/src/commands/deposit-all.ts [hunk 4]
```diff
@@ -227,13 +227,13 @@ export default class DepositAll extends IronfishCommand {
       const sendableOre = confirmedBalance - fee
 
       // terminate condition
-      if (terminate && pendingBalance < sendableOre + fee) {
+      if (terminate || pendingBalance < fee) {
```
Comment (hughy) [inline ironfish-cli/src/commands/deposit-all.ts:230]: The condition here isn't quite right.

It needs to be `&&`: `terminate` is a command-line flag, so if it is set to true then this change would result in termination after the first deposit.

The check here doesn't account for `minDepositOre`: if the pending balance is less than `minDepositOre + fee` then no more deposits can be made.

**Comment by NullSoldier** [comment]
This code has been deleted.

**Pull request closed without merging**